src/coreclr/jit/morph.cpp

```diff
@@ -6490,6 +6490,15 @@ GenTree* Compiler::fgMorphField(GenTree* tree, MorphAddrContext* mac)
                 tree->SetOper(GT_IND);
                 tree->AsOp()->gtOp1 = addr;
 
+                bool plsSpeculative = true;
+                if ((info.compCompHnd->getStaticFieldCurrentClass(symHnd, &plsSpeculative) != NO_CLASS_HANDLE) &&
+                    !plsSpeculative)
+                {
```

Comment:
I suppose this misses the case where the ref-type static is initialized to `null`, presumably those are rare?

Are we able to clear `GTF_ICON_INITCLASS` here?

Reply:
> I suppose this misses the case where the ref-type static is initialized to null, presumably those are rare?

@AndyAyersMS Yeah, but for null we don't need `GTF_IND_INVARIANT` anyway - we'd rather just emit `gtNewIcon(0, TYP_REF)` instead.
I had a quick PR to do that https://github.com/dotnet/runtime/pull/42078

and just in case - if a static-readonly field is null (and the class is initialized statically already) - we won't enter this branch.

> Are we able to clear GTF_ICON_INITCLASS here?

I assume yes, the class is already inited - we can do whatever we want with that IND, the only usage of that flag is loop hoisting